bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -470,10 +470,36 @@ impl Timeline {
         Ok(())
     }
 
+    /// Edits an event from the timeline.
+    ///
+    /// Only works for events that exist as timeline items.
+    ///
+    /// If it was a local event, this will *try* to edit it, if it was not
+    /// being sent already. If the event was a remote event, then it will be
+    /// redacted by sending an edit request to the server.
+    ///
+    /// Returns whether the edit did happen. It can only return false for
+    /// local events that are being processed.
     pub async fn edit(
         &self,
+        item: Arc<EventTimelineItem>,
         new_content: Arc<RoomMessageEventContentWithoutRelation>,
+    ) -> Result<bool, ClientError> {
+        let edit_info = item.0.edit_info().map_err(|err| anyhow::anyhow!(err))?;
```

Comment:
`ClientError::from(err)` doesn't work (to replace `anyhow!`)?

Reply:
Yes, with a new conversion method. I'll add it. We should do a pass at making better errors at the FFI layer some day.